### ISSUE TITLE
Enabling DirectRouting for vxlan on running cluster

### ISSUE THREAD

**Issue by qrpike** [flannel-io/flannel#892]
I have a 16 node working cluster using flannel ( vxlan mode with defaults ) for the kubernetes cni.

All 16 machines are bare metal CoreOS PXE boots, and are directly connected on the same subnet. ( `10.10.2.0/24` ). I see there is a new option for DirectRouting as long as all the nodes are on the same network. 

I read in the docs that you cannot change the flannel *type* without restarting all nodes flanneld, but can you enable DirectRouting ( using etcd ) and not have to experience downtime?

If it does require downtime, should I be able to `systemctl restart flanneld` on all nodes at the same time and it come back up and continue working properly? Or does it require shutting down everything in k8s and restarting them, then starting all pods?

**Comment by tomdee** [comment]
Yes, you should be able to enable DirectRouting and experience minimal. I'm not really sure if existing TCP connections would be OK, I'd need to think about it some more, and you'll need to restart the flannel deamons but they can be done individually. 

Please let us know how you get one and it would be great to get the docs updated with some additional info about this.

**Comment by qrpike** [comment]
So I performed the upgrade last night. I had to set the version to 0.9.1 since I was on 0.8 which didnt actually support DirectRouting. Once I had them all upgraded I tried upgrading 1 host individually which didn't work ( containers couldn't access others where directrouting wasn't enabled yet. ). 

With all services/containers running, I restarted flanneld on all nodes at once ( which causes docker to restart, and all containers to restart ). However once they came back online, everything worked as expected. 

So I guess takeaways are: Must update *all* nodes at one time, there will be minimal downtime for containers to restart. Must be on version 0.9+
